### ISSUE TITLE
[css-contain] To which box does paint containment clip?

### ISSUE THREAD

**Issue by Loirooriol** [w3c/csswg-drafts#1809]
[CSS Contain](https://drafts.csswg.org/css-contain/#containment-paint) says

> If the element does not generate a principal box [...] paint containment has no effect. Otherwise, giving an element paint containment has the following effects:
> - The contents of the element must be clipped to the element’s content box, including both the paint of the descendants and their geometry.

First, it should be "content area of a box" instead of "content box". Then, of which box? An element may generate multiple boxes; for example, a table element generates a principal table wrapper box and an inner table box. I guess it should be the principal box, mentioned earlier, but that's not clear.

**Comment by frivoal** [comment]
Hmm, good question. @FremyCompany What do you think about the table case? @tabatkins what about lists?

**Comment by tabatkins** [comment]
"Principal box" seems probably right on first glance, yeah.

**Comment by FremyCompany** [comment]
The table-wrapper-box vs table-grid-box is a concept for which only those who give it credit will find issues to solve. These issues do not exist in the first place 🤷‍♂️

There is no need for two table boxes. The only "real" table box is what the specs call the "table wrapper box". In this case like in all others that involve anything table-related besides painting, that is the only box we will ever need to consider. 

Unfortunately, the specs were written the other way around, and caused a close-to-infinite confusion.

**Comment by frivoal** [comment]
@FremyCompany In this case, it is related to painting but, if we don't go with the table-wrapper-box, presumably we'll be clipping out table captions, which sounds silly.

@tabatkins If we just pick `Principal box` for things like list items, doesn't that mean that we clip out list markers when `list-style-position` is `outside?  Or is that a feature?

**Comment by FremyCompany** [comment]
@frivoal Yes, totally, the anwser like in all cases is the table wrapper box, the only real box. 

When I said except for painting, I meant its own painting, not its interactions with the outer world through composition. In fact I can be more precise, the only two painting exceptions that I see are interactions between a table and the backgrounds and borders properties.

**Comment by Loirooriol** [comment]
Note that the table-wrapper-box vs table-grid-box is not the only case with multiple boxes. For example, `display: block ruby` generates a principal block container containing an inline-level ruby container.
